homeassistant/components/squeezebox/__init__.py

```diff
@@ -1,31 +1,80 @@
 """The Squeezebox integration."""
 
+from asyncio import timeout
 import logging
 
+from pysqueezebox import Server
+
 from homeassistant.config_entries import ConfigEntry
-from homeassistant.const import Platform
+from homeassistant.const import (
+    CONF_HOST,
+    CONF_PASSWORD,
+    CONF_PORT,
+    CONF_USERNAME,
+    Platform,
+)
 from homeassistant.core import HomeAssistant
+from homeassistant.exceptions import ConfigEntryNotReady
+from homeassistant.helpers.aiohttp_client import async_get_clientsession
 
-from .const import DISCOVERY_TASK, DOMAIN, PLAYER_DISCOVERY_UNSUB
+from .const import (
+    CONF_HTTPS,
+    DATA_SERVER,
+    DISCOVERY_TASK,
+    DOMAIN,
+    PLAYER_DISCOVERY_UNSUB,
+    STATUS_API_TIMEOUT,
+    STATUS_QUERY_LIBRARYNAME,
+    STATUS_QUERY_UUID,
+)
 
 _LOGGER = logging.getLogger(__name__)
 
 PLATFORMS = [Platform.MEDIA_PLAYER]
 
-
 async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
-    """Set up Squeezebox from a config entry."""
+    """Set up an LMS Server from a config entry."""
+    config = entry.data
+    session = async_get_clientsession(hass)
+    _LOGGER.debug("Reached async_setup_entry for host=%s(%s)", config[CONF_HOST],entry.entry_id)
+
+    username = config.get(CONF_USERNAME)
+    password = config.get(CONF_PASSWORD)
+    https = config.get(CONF_HTTPS, False)
+    host = config[CONF_HOST]
+    port = config[CONF_PORT]
+
+    hass.data.setdefault(DOMAIN, {})
+    entry.runtime_data = {}
+
+    lms = Server(session, host, port, username, password, https=https)
+    _LOGGER.debug("LMS object for %s", lms)
+
+    try:
+       async with timeout(STATUS_API_TIMEOUT):
+          status = await lms.async_query("serverstatus","-","-","prefs:libraryname")
+    except Exception as err:
+       raise ConfigEntryNotReady(f"Error communicating config not read for {host}") from err
+
+    if not status:
+       raise ConfigEntryNotReady(f"Error Config Not read for {host}")
+    _LOGGER.debug("LMS Status for setup  = %s", status)
+
+    lms.uuid = status[STATUS_QUERY_UUID]
+    lms.name = (STATUS_QUERY_LIBRARYNAME in status and status[STATUS_QUERY_LIBRARYNAME]) and status[STATUS_QUERY_LIBRARYNAME] or host
+    _LOGGER.debug("LMS %s = '%s' with uuid = %s ", lms.name, host, lms.uuid)
+
+    entry.runtime_data[DATA_SERVER] = lms
```

Comment:
Should we only store the lms in there?

Reply:
Other items will be added later

Reply:
But we can add that in a follow up right?

Like now we can type the ConfigEntry with the LMS, and the moment you need to add more stuff, you introduce a `@dataclass` with the other properties you want to store

Reply:
its now used in media_player only, so put this is an ConfigEntry class for LMS? look at Air Gradient right?

Reply:
If it's only used in media player and nowhere else, just keep it in media player